### ISSUE TITLE
Interrupting the kernel does not work on Windows

### ISSUE THREAD

**Issue by amilsted** [JuliaLang/IJulia.jl#503]
I am running IJulia v1.4.1 on Julia v0.5.0 under Windows 10. 

I am unable to interrupt the kernel during a computation with Julia running on Windows, despite being able to interrupt the same operations without issue with Julia running on Linux. This is not a new issue - I have been using IJulia for about a year (mostly with Julia v0.4.x) and have never so far found the kernel interrupt feature to work on Windows.

In summary: The "interrupt" option in the "Kernel" menu appears to do nothing except flash the message "Interrupting the kernel" when the kernel is running on Windows.

**Comment by amilsted** [comment]
Presumably related to quinnj/Sublime-IJulia#33.

**Comment by stevengj** [comment]
Jupyter is supposed to be sending a SIGINT to the kernel process.  @minrk, does it do something different on Windows, since signals work differently there?

**Comment by minrk** [comment]
Yes, signals are weird on Windows, so we send a Windows Event [here](https://github.com/jupyter/jupyter_client/blob/5.0.0/jupyter_client/manager.py#L404), created [here](https://github.com/jupyter/jupyter_client/blob/5.0.0/jupyter_client/win_interrupt.py). The event is passed as the JPY_INTERRUPT_EVENT environment variable (an integer). IPython's handling of this is [here](https://github.com/ipython/ipykernel/blob/4.5.2/ipykernel/parentpoller.py#L43).

I have a vague recollection that this was because when the kernel is given a new process group, it can no longer be interrupted.

I opened https://github.com/jupyter/jupyter_client/issues/244 to document this, and possibly revert to sending plain signals on Windows.

**Comment by janschulz** [comment]
This might be related: https://github.com/jupyter/jupyter_client/issues/104

**Comment by filmor** [comment]
Since I had the same issue with my Erlang Jupyter kernel, I've added an `interrupt_mode` configuration option that allows you to receive a message instead of a SIGINT (https://github.com/jupyter/jupyter_client/pull/294). I'll look into implementing this in IJulia.

**Comment by filmor** [comment]
I did (well, actually only after you pinged and I looked into it again ;)), it works for me and is pending review.
